### PR TITLE
Show all domains in the permission list

### LINKED ISSUE #7634: Show all domains in list of permissions
There is no way for a user to know what all domains can an extension access data from if the host permissions >= 4.

Consider this extension https://addons.mozilla.org/en-US/firefox/addon/google-search-result-shortcuts
It says it can access data from google.com, google.ad, google.ae and **`190 other domains`** - This information is kind of really insufficient and vague for a user.
I am sure we dont want to put all 190 domains in product page, but may be a popup or `see more` kind of stuff may help.

image

### PR DESCRIPTION
**Pull request by mirefly**
Fixes #7634
It looks like `ShowMoreCard` is good for this case. Thanks @willdurand for the suggestion.

In this patch, we
- Changed `Card` to `ShowMoreCard` in `PermssionsCard`, and listed all domains.
- moved all of "Access you data for *"(sites/domains) to the end of the list, so that we do not need to scroll down to see other permissions, especially when there are many sites/domains in this list.

New UI: http://localhost:3000/en-US/firefox/addon/all-of-the-permissionss/
image

### CODE REVIEW

**Comment by willdurand** [comment]
> We might want to move all of "Access you data for sites *" to the end of the list, so that we do not need to scroll down to see other permissions, especially when there are many sites/domains in this list.

This sounds like a very good idea.

**Comment by mirefly** [comment]
I added c27b66b to move the site/host permissions to the end of the list. This patch is ready for a review.

File: src/ui/components/HostPermissions/index.js [hunk 1]
```diff
@@ -108,20 +108,6 @@ export class HostPermissionsBase extends React.Component<InternalProps> {
...
     }
-    if (permissions.length > 4) {
-      // Replace the final individual permission with a "too many" permission.
-      hostPermissions[3] = (
-        <Permission
-          type="hostPermission"
-          description={this.getPermissionString({
-            messageType,
-            param: permissions.length - 3,
-            multiple: true,
```
Comment (willdurand) [inline src/ui/components/HostPermissions/index.js:119]: So do we actually need this `multiple` logic in `getPermissionString()`?

File: src/ui/components/ShowMoreCard/index.js [hunk 2]
```diff
@@ -99,11 +100,14 @@ export class ShowMoreCardBase extends React.Component<InternalProps> {
...
   truncateToMaxHeight = (contents: HTMLElement | null) => {
-    const { uiState } = this.props;
+    const { maxHeight, uiState } = this.props;
     if (contents) {
       // If the contents are short enough they don't need a "show more" link; the
       // contents are expanded by default.
-      if (uiState.showAllContent && contents.clientHeight >= MAX_HEIGHT) {
+      if (
+        uiState.showAllContent &&
+        contents.clientHeight >= (maxHeight || DEFAULT_MAX_HEIGHT)
```
Comment (willdurand) [inline src/ui/components/ShowMoreCard/index.js:109]: Shouldn't we assign `maxHeight` to `DEFAULT_MAX_HEIGHT` in the `defaultProps`?

File: tests/unit/amo/components/TestPermissionUtils.js [hunk 3]
```diff
@@ -105,21 +105,21 @@ describe(__filename, () => {
...
         'nativeMessaging',
         'Exchange messages with programs other than Firefox',
       );
       // Named permissions in alphabetical order.
-      expectPermission(result[2], 'bookmarks', 'Read and modify bookmarks');
-      expectPermission(result[3], 'tabs', 'Access browser tabs');
+      expectPermission(result[1], 'bookmarks', 'Read and modify bookmarks');
+      expectPermission(result[2], 'tabs', 'Access browser tabs');
+      // HostPermissions component.
+      expect(result[3].props.permissions).toHaveLength(2);
```
Comment (willdurand) [inline tests/unit/amo/components/TestPermissionUtils.js:118]: I suppose this isn't really needed if we use `toEqual` with an array below.

File: src/ui/components/HostPermissions/index.js (hunk 1 above)
Comment (mirefly) [reply src/ui/components/HostPermissions/index.js:119]: We do not need it any more.

**Comment by mirefly** [comment]
@willdurand Thanks for reviewing and helpful comments. I made the changes requested.

**Comment by willdurand** [comment]
Thanks!

**Pull request merged**